### ISSUE TITLE
🐛 Buggy side-effect caused by previous fix that made `biome.json(c)` un-ignore-able

### ISSUE THREAD

**Issue by KemingHe** [biomejs/biome#2990]
### Environment information

```block
Platform:
  CPU Architecture:             x86_64
  OS:                           linux

Environment:
  BIOME_LOG_DIR:                unset
  NO_COLOR:                     unset
  TERM:                         "xterm-256color"
  JS_RUNTIME_VERSION:           "v22.2.0"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         "pnpm/9.1.2"

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           false
  Linter disabled:              false
  Organize imports disabled:    false
  VCS disabled:                 false

Workspace:
  Open Documents:               0
```

### What happened?

To replicate the issue (after setting up the same env as mine):

1. In your `biome.json`, specify:

```javascript
{
  "$schema": "https://biomejs.dev/schemas/1.7.3/schema.json",
  ...
  "json": {
      "formatter": {
        "trailingCommas": "all"
      }
    },
  ...
}
```

2. Run `npx biome check --apply .` (or yarn, or npmp, or predefined script in `package.json`);

3. `biome.json` will be formatted with trailing commas, which breaks the json standard for a root level config data file. To demonstrate, run `npx biome check --apply .` again and receive this error:

```bash
✖ Expected a property but instead found '}'.
  
    51 │     },
    52 │   },
  > 53 │ }
       │ ^
    54 │ 
```

4. Furthermore,  merge #1531 specified `biome.json` to be **un-ignore-able**, meaning that while the rest of the `**/*.json` files can be ignored or fine-grained controlled to add/not add commas (i.e. `package.json`), **`biome.json`** will always be incorrectly formatted.

### Expected result

`biome.json` should be (by design, not by user choice) untouched by the trailing comma formatting option. 

I would further suggest that user would not be able to specify to apply formatting to the file to avoid confusion.

In short:
**Given** the `trailingCommas` formatter for json is set to `all`
**When** running `npx biome check --apply .`
**Then** `biome.json` should not be affected

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by KemingHe** [comment]
For those come across here looking for a workaround, here's what you need to do:

```javascript
"json": {
    "formatter": {
      "trailingCommas": "none"
    }
  },
```

And keep the discussion going so our wonderful maintainers can see this issue. ❤️

**Comment by dyc3** [comment]
I'd like to give this a shot.

**Comment by nhedger** [comment]
> I'd like to give this a shot.

Thanks, assigned!
